src/regex.rs

```diff
@@ -10,6 +12,20 @@ crate::binding_utils::gen_proto!(Regex, __str__);
 
 #[pymethods]
 impl Regex {
+    #[classmethod]
+    fn from_file(_cls: &PyType, pattern_file_content: &str) -> PyResult<Self> {
+        libparsec::types::Regex::from_file(Path::new(pattern_file_content))
+            .map(Self)
+            .map_err(|err| PyValueError::new_err(err.to_string()))
+    }
+
+    #[classmethod]
+    fn from_raw_regexes(_cls: &PyType, raw_regexes: Vec<&str>) -> PyResult<Self> {
```

Comment:
```suggestion
    fn from_raw_regexes(_cls: &PyType, raw_regexes: &[str]) -> PyResult<Self> {
```

Reply:
If we do so, we will get a compile error because the size of `str` can't be known at compile time. Unfortunately the type `&[&str]` does not implement the `PyClass` trait :(